pkg/web/initclient/init_http_handler_test.go

```diff
@@ -0,0 +1,99 @@
+package initclient
+
+import (
+	"github.com/aerogear/mobile-security-service/pkg/helpers"
+	"encoding/json"
+	"net/http"
+	"net/http/httptest"
+	"strings"
+	"testing"
+
+	"github.com/google/uuid"
+
+	"github.com/aerogear/mobile-security-service/pkg/models"
+
+	"github.com/aerogear/mobile-security-service/pkg/web/apps"
+	"github.com/labstack/echo"
+)
+
+func TestHTTPHandler_InitClientApp(t *testing.T) {
+	type fields struct {
+		appsService apps.Service
+	}
+	type args struct {
+		device models.Device
+	}
+	tests := []struct {
+		name           string
+		fields         fields
+		args           args
+		wantStatusCode int
+	}{
+		{
+			name: "A 400 Bad Request should be returned when request body is missing device ID",
+			args: args{
+				device: helpers.getMockDeviceWithMissingDeviceID(),
```

Comment:
Should be better you have just one mock and update it as suggested. 
It is better in this way than before, however, the code still not centralized. 

See how it will be with the suggestion.
**In the helper you will have just one mock for the device**
Then in the test func you can do as.

```go
func TestHTTPHandler_InitClientApp(t *testing.T) {
deviceWithMissingDeviceID:= helpers.GetMockDevice()
deviceWithMissingDeviceID.deviceID = ""

deviceWithMissingVersion:= helpers.GetMockDevice()
deviceWithMissingDeviceID.version = ""

{...}

     tests := []struct {
		name    string
		fields  fields
		device model.Device
		wantErr error
	}{
		{
			name: "A 400 Bad Request should be returned when request body is missing device ID",
			device: deviceWithMissingDeviceID
                         wantStatusCode: 400,
		},
		{
			name: "A 400 Bad Request should be returned when request body is missing version",
			device: deviceWithMissingVersion
                         wantStatusCode: 400,
		},
	}

{...}
```
Is it make sense? 
PS.: Anyway, it is better than before.